File: Startupba/UI/startupba_desktop/lib/model/donation.dart
import 'package:json_annotation/json_annotation.dart';

part 'donation.g.dart';

@JsonSerializable()
class Donation {
  final int id;
  final int startupId;
  final String startupName;
  final int userId;
  final String userName;
  final double amount;
  final String? message;
  final String status;
  final DateTime createdAt;
  final DateTime? completedAt;

  Donation({
    this.id = 0,
    this.startupId = 0,
    this.startupName = '',
    this.userId = 0,
    this.userName = '',
    this.amount = 0,
    this.message,
    this.status = '',
    required this.createdAt,
    this.completedAt,
  });

  factory Donation.fromJson(Map<String, dynamic> json) =>
      _$DonationFromJson(json);
  Map<String, dynamic> toJson() => _$DonationToJson(this);
}

File: Startupba/Startupba.Services/Database/Donation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Startupba.Services.Database
{
    /// <summary>
    /// A donation (investment) made by a user to a startup.
    /// Created as "Pending" and marked "Completed" once payment is confirmed.
    /// </summary>
    public class Donation
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int StartupId { get; set; }

        [Required]
        public int UserId { get; set; } // FK to User (donor / investor)

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        /// <summary>
        /// Optional message of support shown to the founder.
        /// </summary>
        [MaxLength(500)]
        public string? Message { get; set; }

        /// <summary>
        /// "Pending", "Completed", "Failed"
        /// </summary>
        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "Pending";

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public DateTime? CompletedAt { get; set; }

        // Navigation properties
        [ForeignKey("StartupId")]
        public Startup Startup { get; set; } = null!;

        [ForeignKey("UserId")]
        public User User { get; set; } = null!;
    }
}
